src/treq/client.py

```diff
@@ -98,6 +96,11 @@ def deliverBody(self, protocol):
             self._waiters.append(protocol)
 
 
+class _NoJsonData():
+    """Dummy class to identify data that is not Json."""
```

Comment:
What about a generic `NOTHING` class/instance to be used when `None` is a valid value to pass in but we still want to know when nothing was passed?  Mostly this functionality just seems totally independent of the parameter being meant to receive JSON or anything else.  I haven't actually looked around treq or twisted for something existing like this.

I don't know about bothering to make it a singleton but...

https://github.com/python-attrs/attrs/blob/eda9f2de6f7026039c3eee80297f0c95382d51d8/src/attr/_make.py#L49-L67

Reply:
@twm's proposal is simpler for this case.

Reply:
It is.  One of the points of having a class/instance is to have an identifiable `repr()` in case it shows up in tracebacks etc and `object()` doesn't provide that.

Reply:
Yup, but I'd think we don't care here, do we? Nothing should be interacting with that besides the line checking if we want json or not.

Reply:
It is true that you don't need easy debugging until you need to debug something... Though you only need to debug when the unexpected happens.  :]  But, no biggy.  A global `NOTHING` object for treq could easily be argued to be a separate PR if needed.